src/coreclr/jit/importercalls.cpp

```diff
@@ -5014,60 +5014,52 @@ GenTree* Compiler::impIntrinsic(CORINFO_CLASS_HANDLE    clsHnd,
 #endif // FEATURE_HW_INTRINSICS
 
 #ifdef TARGET_RISCV64
-                GenTree* op2 = impImplicitR4orR8Cast(impPopStack().val, callType);
-                GenTree* op1 = impImplicitR4orR8Cast(impPopStack().val, callType);
-
-                if (isNative)
+                if (!isMagnitude)
```

Comment:
Is this check necessary? Should it be an assert instead since you're no longer handling `MaxMagnitude` and it shouldn't reach here?